Feature - 모임 상세 뷰 분기처리(모임 정보, 멤버 리스트)

yangsubinn opened the pull request:
## 관련 이슈
<!--
이슈 번호 #000 작성  
ex) close #1 
--> 
close #86

## 내용
<!--
로직 설명  
--> 

**1. 모임명 dynamic width**
  - 너비가 190px 초과하는 경우 줄바꿈 + 옆에 스탬프 위치 모임명 바로 옆
  - maxWidth 사용하면 간단하게 할 수 있을 줄 알았는데 maxWidth를 잡는 순간 너비 자체가 190으로 확정되고 그 이상이 되면 줄바꿈 되는 형태더라구요..
  - 별다른 방법이 생각나지 않아서 9글자 이상 넘어가는 경우 maxWidth를 적용하는 식으로 했습니다. 
  - 190이 아니라 200으로 한 것은, 폰트 자간이 피그마 디자인과 정확하게 맞지 않아서 그런지 디자인 예시로 있던 글자를 넣었을때 동일하게 줄바꿈하도록 하려면 197 이상이어야 하더라구요. 넉넉하게 200으로 반올림했습니다.

| 한줄짜리 | 두줄이상 |
|--|--|
|![스크린샷 2023-08-08 오전 12 31 29](https://github.com/dnd-side-project/dnd-9th-2-ios/assets/81167570/5a979e6c-c73f-4fd0-9bcf-076941355493) | ![스크린샷 2023-08-08 오전 12 31 16](https://github.com/dnd-side-project/dnd-9th-2-ios/assets/81167570/6f551e5a-4cc4-4e8b-844c-9a09783724f2) |


**2. 스탬프 분기처리 (약속 확정 / 완료)**
  - MeetingStatus를 ready / progress / confirmed / completed 로 분리한 뒤, confirmed 또는 completed인 경우에만 각각 스탬프 보이도록 했습니다.


**3. 메모 분기처리**

| 메모 O | 메모 X |
|--|--|
|![스크린샷 2023-08-08 오전 12 45 49](https://github.com/dnd-side-project/dnd-9th-2-ios/assets/81167570/4b50c5b1-c9be-4271-8036-d77134fbe44f) | ![스크린샷 2023-08-08 오전 12 47 17](https://github.com/dnd-side-project/dnd-9th-2-ios/assets/81167570/2c4aad2a-74cf-4651-a25f-fa2fde6a68a8) |


**4. 멤버 리스트 인증 여부 동그라미, 방장, 긴급 버튼 할당자 분기처리**
  - 임시로 경우의 수 다 넣어두었습니다.
![스크린샷 2023-08-08 오전 12 49 09](https://github.com/dnd-side-project/dnd-9th-2-ios/assets/81167570/4797e785-20e3-4da1-b9c6-9074c7faa578)


**5. 에셋 추가**
  - 기본 프로필 이미지


### 이미지, 영상
<!--
필요시 스크린샷 첨부
--> 
생략


## 리뷰어가 확인할 사항
<!--
중점적으로 봐주면 좋을 사항  
ex) ㅇㅇㅇㅇ하려고 Combine 사용했는데 제대로 사용하고 있는건가요?
--> 
dynamicWidth 구현할 수 있는 다른 방법이 있다면 알려주세요 .. 

## 기타
<!--
레퍼런스 혹은 패키지 설치 등
-->
아 ..... 커밋메세지에 이슈 번호 잘못 썼네요 ...................... 죄송합니다 .......

yangsubinn committed:
Fix #86 - MeetingStatus 수정, 모임 상세 스탬프 분기처리

yangsubinn committed:
Merge: develop pull
Merge branch 'develop' of https://github.com/dnd-side-project/dnd-9th-2-ios into feature/meetingDetail-meetingInfo-ui

yangsubinn committed:
Feat #86 - 모임 상세 메모 분기처리

yangsubinn committed:
Feat #88 - 모임 상세 멤버 리스트 분기처리

- 기본 프로필 이미지 에셋 추가
- ProfileBadgeView 추가 (방장/버튼할당자)
- CircleProfileView 기본 이미지 추가
- Member isButtonOwner 추가

yangsubinn committed:
Fix #88 - 모임 상세 타이틀 maxWidth 수정

never-better commented on Baggle/Baggle/Core/Models/MeetingDetail.swift:
```diff
@@ -13,6 +13,7 @@ struct Member: Identifiable, Hashable {
     let name: String // 유저 이름
     let profileURL: String // 유저 프로필 이미지 URL
     let isOwner: Bool // 방장 여부
+    let isButtonOwner: Bool // 긴급 버튼 할당자 여부
```
p4: 서버랑 통일성있게 `isAuthorization` 식으로 가도 좋을 것 같아요.

never-better commented on Baggle/Baggle/Features/MeetingDetail/Cell/FeedListCell.swift:
```diff
@@ -34,10 +35,12 @@ struct FeedListCell: View {
-                    .cornerRadius(12)
             } placeholder: {
                 Color.gray
+                    .frame(width: UIScreen.main.bounds.width-40,
```
p4: Stack에 Spacer()를 쓰는게 더 SwiftUI스러울 것 같아요.

yangsubinn replied:
Spacer를 사용하면 placeholder 크기가 안잡혀서 대신 image만 적용하던 aspectRatio 사용해서 수정하겠습니다

<img src="https://github.com/dnd-side-project/dnd-9th-2-ios/assets/81167570/b156a64a-5b58-4a17-8c9f-3e05557f84d3" width=200 >

```
AsyncImage(url: URL(string: feed.feedImageURL)) { image in
                image
                    .resizable()
            } placeholder: {
                Color.gray
            }
            .aspectRatio(1.0, contentMode: .fill)
            .cornerRadius(12)
            .clipped()
```

never-better replied:
Kingfisher를 빠르게 설치하고 사용할까요??

yangsubinn replied:
오오 #87 해당 이슈에서 kingfisher 추가하고 asyncImage 사용한 부분 수정해두겠습니다!

never-better commented on Baggle/Baggle/Features/MeetingDetail/MeetingDetailView.swift:
```diff
@@ -125,57 +125,64 @@ extension MeetingDetailView {
-
-                // dynamin width 수정
+                
                 Text("\(data.name)")
```
저도 예시로 해봤는데 잘 안 되더라구요...
그나마 Text Size 측정 가능한게
1. NSAttributedString 사용하기
2. Geometry Reader 사용하기
인 것 같습니다. 

아래는 NSAttributedString 사용해 String 사이즈 측정한 거 입니다. 양 옆에 패딩이 남지않아 더 깔끔하게 맞춰줄 수 있을 것 같습니다.

never-better replied:
```swift
struct MaxWidth: View {
    
    let text = "안녕하세요안녕하세요안녕하세요"
    var width: CGFloat {
        return String(text.prefix(10)).widthOfString(
                    usingFont: UIFont.systemFont(ofSize: 17, weight: .bold)
                )
    }
    
    var body: some View {
    
        HStack {
            Text("📌")
            
            Text(text)
                .font(.system(size: 17).bold())
                .frame(maxWidth: text.count > 10 ? width : .none)
                .background(Color.red)
                
            Image(systemName: "ruler")
                .resizable()
                .scaledToFill()
                .frame(width: 50, height: 10)
                
            Spacer()
        }
        
    }
}
```

never-better replied:
![스크린샷 2023-08-08 오전 10 13 52](https://github.com/dnd-side-project/dnd-9th-2-ios/assets/71776532/476cc3a2-4033-433b-a436-fde2f0e39299)

yangsubinn replied:
혹시 text 개수로 자르게 되면 어차피 width에 상관없이 190-200이라는 한정된 값이 지정해주게 되는데, 텍스트의 width가 필요할까요?

nsattributedstring 사용해서 string 사이즈 측정값 사용하려면 text.count로 자르지 말고 아래처럼 사용하는건 어떤가요?

```
extension String {
    var width: CGFloat {
        let attributedString = NSAttributedString(string: self, attributes: [NSAttributedString.Key.font: UIFont.systemFont(ofSize: 22.0)])
        let stringSize = attributedString.size()
        let widthOfString = stringSize.width

        return widthOfString
    }
}

Text("수빈님네 집들이집들이 집들이집들이집")
                    .frame(maxWidth: data.name.width > 200 ? 200 : .none, alignment: .leading)
```
![스크린샷 2023-08-08 오전 11 36 43](https://github.com/dnd-side-project/dnd-9th-2-ios/assets/81167570/fdc46af2-7ac4-4580-b42d-12c1b375960c)
![스크린샷 2023-08-08 오전 11 37 01](https://github.com/dnd-side-project/dnd-9th-2-ios/assets/81167570/3b91a0bf-1f5d-4ad8-868f-5ded97ffe13f)

never-better replied:
![Group 585](https://github.com/dnd-side-project/dnd-9th-2-ios/assets/71776532/89a21bee-1070-41ef-8477-a153c99d224f)

글자마다 크기가 달라 제목에 따라 파란색 공간 크기가 달라져서 디자인이 맞추기 어렵지 않을까 생각했었습니다.

큰 상관 없으면 저는 고정 width도 좋습니다!

never-better commented on Baggle/Baggle/Features/Tab/Home/HomeFeature.swift:
```diff
@@ -14,21 +14,22 @@ import KakaoSDKShare
         switch self {
-        case .ongoing, .dday: return "예정된 약속"
-        case .complete: return "지난 약속"
+        case .ready, .progress, .confirmed: return "예정된 약속"
```
Q: title은 어디서 사용되는 프로퍼티 인가요?? `.confirmed`는 문맥상 확정된 약속이 되어야할 것 같은데 어디서 사용되는지를 못 찾게어서 모르겠어요.

yangsubinn replied:
![스크린샷 2023-08-08 오전 11 13 53](https://github.com/dnd-side-project/dnd-9th-2-ios/assets/81167570/73fa67f0-69a3-4eb4-ab85-bd51d39230d0)
세그먼트 피커에서 사용되는 타이틀입니다
약속 완료 여부만 기준으로 하기 때문에 완료 전은 모두 예정된 약속에 포함시켰습니다
조금 헷갈릴 여지가 있는 것 같아서 segmentTitle로 이름 변경해두겠습니다

never-better replied:
확인했습니다!

never-better commented on Baggle/Baggle/Features/Tab/Home/HomeFeature.swift:
```diff
@@ -14,21 +14,22 @@ import KakaoSDKShare
 import KakaoSDKTemplate
 
 enum MeetingStatus {
```
p3: 파일 분리 해주세요

yangsubinn committed:
Chore #91 - 코드리뷰 반영 수정

- MeetingStatus 수정 반영 안된 부분 수정
- String+ width 추가
- 네이밍 수정
- MeetingStatus 파일 분리

yangsubinn committed:
Merge: develop pull

yangsubinn committed:
Merge: develop pull
Merge branch 'develop' of https://github.com/dnd-side-project/dnd-9th-2-ios into feature/meetingDetail-meetingInfo-ui
